src/Cardano/Types.hs

```diff
@@ -187,19 +171,19 @@ isValidCoin c = c >= minBound && c <= maxBound
                                     UTxO
 -------------------------------------------------------------------------------}
 
-newtype UTxO = UTxO
-    { getUTxO :: Map TxIn TxOut }
+newtype UTxO i = UTxO
+    { getUTxO :: Map i Coin }
```

Comment:
Perhaps use a different type variable here (such as `out` or `o`), as this represents an unspent transaction **output** set.